### PR TITLE
Added input type validation to the beginning of the histogram function.

### PR DESCRIPTION
**Pull request by thomasj-s**
# Histogram function input type validation

[Test endpoint: `viz/histograph`](https://trello.com/c/l4arGU8p)

---

### Location:

app/api/utils/visualizations/histogram.py
app/tests/test_visualizations.py

---

### Details:

1. This was added as redundancy to insure that the input types match the specified types outlined in the 'HistogramRequest' BaseModel.  Now, if there are any input discrepancies, the function will return the error message before the function finishes running. 

2. Added docstrings and comments to clarify the functionality of the input validators in the histogram function.  Also updated test_visualizations.py with test cases for the histogram function.

---

### Committed Features:
1. Added input type validation to the beginning of the histogram functions.
2. Added docstrings and comments to clarify the functionality of the input validators added in the first commit.
3. Added tests cases to test_visualizations.py testing the input and output of the histogram function.

---

### Any Issues:
1. No issues.

### CODE REVIEW

File: app/utils/visualizations/histogram.py [hunk 1]
```diff
@@ -5,6 +5,68 @@
 # Imports
 import plotly.express as px
 
+'''
+The following functions serve as validtion on the HistogramRequest
+class outlined in app/api/models.py. They test for input specifications
+that are specific to Story Squad; such as the grade level being between
+8-12.
+'''
```
Comment (jmmiddour) [inline app/utils/visualizations/histogram.py:14]: I agree with adding the edge cases to this file for testing but I do not think they need to be in function format. You did a lot of work here and it looks great but the actual testing logic is what we are writing in the test file. If you want to get with me, I can explain my thoughts on this in more detail. This was a great idea though if we did not have the testing file set up.

Comment (jmmiddour) [reply app/utils/visualizations/histogram.py:14]: Thank you for clarifying why you added the functions in this way.

**Review by jeremy-jewett — DISMISSED** [review]
This was quite thoroughly done.  Except for the commented code, nicely done!

**Review by jmmiddour — DISMISSED** [review]
The comments make it easier to understand what is happening with this code. You did a great job giving explanations on your code and with the testing logic.

**Review by ToddDiFronzo — DISMISSED** [review]
Looks like a lot in here but from what I can follow, it looks good.

**Review by jproctor-rebecca — CHANGES_REQUESTED** [review]
please resolve merge conflicts

**Review by jmmiddour — APPROVED** [review]
Merge conflicts should be cleared now and this is ready to be merged.

**Review by jrslagle — APPROVED** [review]

File: app/utils/visualizations/histogram.py [hunk 2]
```diff
@@ -9,43 +9,60 @@
...
     return error
 
+# Checks that the Student's score is of type 'int'.
 def student_score_type_test(student_info):
   if type(student_info[2]) != int:
     error = 'Student\'s score is not of type \'integer\'.'
     return error
 
 def input_error_results(grade_list, student_info):
+  '''
```
Comment (jproctor-rebecca) [inline app/utils/visualizations/histogram.py:60]: I really appreciate how well commented your code is.  Thank you.

File: app/utils/visualizations/histogram.py [hunk 3]
```diff
@@ -83,7 +100,7 @@ def histogram(grade_list, student_info):
         Output: Plotly JSON that can be passed to the web to display on the
         parent dashboard
     """
-    # This returns input error messages if any exist.
+    # This returns input errors if any exist and stops the function.
```
Comment (jproctor-rebecca) [inline app/utils/visualizations/histogram.py:103]: This clarification adds value.  Thank you for the revision.

**Review by jproctor-rebecca — DISMISSED** [review]
